src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs

```diff
@@ -291,7 +299,8 @@ private void ChangeHtmExtensionToHtmlIfRequired(string reportDirectory, IExecuti
                 }
                 catch (Exception ex)
                 {
-                    executionContext.Warning(StringUtil.Loc("RenameIndexFileCoberturaFailed", htmIndexFile, defaultIndexFile, _codeCoverageTool, ex.InnerException.ToString()));
+                    Exception exToReport = ex.InnerException == null ? ex : ex.InnerException;
+                    executionContext.Warning(StringUtil.Loc("RenameIndexFileCoberturaFailed", htmIndexFile, defaultIndexFile, _codeCoverageTool, exToReport.ToString()));
```

Comment:
you can just add a comment explains it, otherwise it's little bit confuse why we always prefer inner exception.